src/contracts/Output/Generator.php

```diff
@@ -75,39 +100,65 @@ protected function checkStartDocument($data)
 
     /**
      * End document.
-     *
-     * Returns the generated document as a string.
-     *
-     * @param mixed $data
-     *
-     * @return string
      */
-    abstract public function endDocument($data);
+    abstract public function endDocument(mixed $data): string;
 
     /**
      * Check end document.
-     *
-     * @param mixed $data
      */
-    protected function checkEndDocument($data)
+    protected function checkEndDocument(mixed $data): void
     {
         $this->checkEnd('document', $data);
     }
 
     /**
      * Start object element.
-     *
-     * @param string $name
-     * @param string $mediaTypeName
      */
-    abstract public function startObjectElement($name, $mediaTypeName = null);
+    public function startObjectElement(string $name, ?string $mediaTypeName = null): void
+    {
+        $this->checkStartObjectElement($name);
+
+        $this->isEmpty = false;
+
+        $mediaTypeName ??= $name;
+
+        $object = new Json\JsonObject($this->json);
+
+        if ($this->json instanceof Json\ArrayObject || $this->json instanceof Data\ArrayList) {
+            $this->json->append($object);
+            if ($this->json instanceof Data\ArrayList) {
+                $this->json->setName($name);
+            }
+            $this->json = $object;
+        } else {
+            $this->json->$name = $object;
+            $this->json = $object;
+        }
+
+        $this->attribute('media-type', $this->getMediaType($mediaTypeName));
+    }
+
+    /**
+     * End object element.
+     */
+    public function endObjectElement(string $name): void
+    {
+        $this->checkEndObjectElement($name);
+
+        if ($this->json->getParent() === null) {
+            throw new \LogicException(sprintf(
+                'Parent element at %s cannot be `null`.',
+                __METHOD__,
+            ));
```

Comment:
If you make `getParent()` part of the interface, you can declare that trying to `getParent()` without one would give this exception.